controllers/serveController.js

```diff
@@ -197,57 +90,47 @@ module.exports = {
         });
     });
   },
-  serveOrShowAsset (fileInfo, extension, method, headers, originalUrl, ip, res) {
-    // add file extension to the file info
-    if (extension === 'gifv') {
-      fileInfo['fileExt'] = 'gifv';
-    } else {
-      fileInfo['fileExt'] = fileInfo.fileName.substring(fileInfo.fileName.lastIndexOf('.') + 1);
-    }
-    // add a record to the stats table
-    postToStats(method, originalUrl, ip, fileInfo.name, fileInfo.claimId, 'success');
-    // serve or show
-    switch (method) {
-      case SERVE:
-        serveFile(fileInfo, res);
-        sendGoogleAnalytics(method, headers, ip, originalUrl);
-        return fileInfo;
-      case SHOWLITE:
-        return db.Claim.resolveClaim(fileInfo.name, fileInfo.claimId)
-         .then(claimRecord => {
-           fileInfo['title'] = claimRecord.title;
-           fileInfo['description'] = claimRecord.description;
-           showFileLite(fileInfo, res);
-           return fileInfo;
-         })
-         .catch(error => {
-           logger.error('throwing serverOrShowAsset SHOWLITE error...');
-           throw error;
-         });
-      case SHOW:
-        return db.Claim
-          .getShortClaimIdFromLongClaimId(fileInfo.claimId, fileInfo.name)
-          .then(shortId => {
-            fileInfo['shortId'] = shortId;
-            return db.Claim.resolveClaim(fileInfo.name, fileInfo.claimId);
-          })
-          .then(resolveResult => {
-            logger.debug('resolve result >>', resolveResult.dataValues);
-            fileInfo['thumbnail'] = chooseThumbnail(resolveResult, DEFAULT_THUMBNAIL);
-            fileInfo['title'] = resolveResult.title;
-            fileInfo['description'] = resolveResult.description;
-            if (resolveResult.certificateId) { fileInfo['certificateId'] = resolveResult.certificateId };
-            if (resolveResult.channelName) { fileInfo['channelName'] = resolveResult.channelName };
-            showFile(fileInfo, res);
-            return fileInfo;
-          })
-          .catch(error => {
-            logger.error('throwing serverOrShowAsset SHOW error...');
-            throw error;
-          });
+  getLocalFileRecord (claimId, name) {
+    return db.File.findOne({where: {claimId, name}})
+      .then(file => {
+        if (!file) {
+          return NO_FILE;
```

Comment:
Is there a reason we don't just use something like `null` to mean no record found? This goes for here and a few other places.

Reply:
I'm elevating this out of the model layer, but keeping at the controller level